### DISCUSSION TITLE
Batch run commands against clusters

### DISCUSSION THREAD

**Discussion by richardcase** [fidelity/kconnect#227] [Ideas] [answered]
Do you think it would be a good idea to enable batch running of kubectl commands against a list of clusters?

Something like this:

`kconnect run --alias cluster1,cluster2 --commands-file commands.txt --output json`

And the file could contain:

```text
kubectl get pods -A
```

**Comment by richardcase** [comment, marked as answer]
@ramnar @rajarajanpsj @seankelly001 - what do you think?

**Reply by seankelly001** [reply]
Think this is a great idea, would make kconnect very powerful!

1 option could be to use a yaml file and run it like so:
kconnect run -f run.yaml --password mypassword123

apiVersion: kconnect.fidelity.github.com/v1alpha1
kind: RunConfig
metadata:
	name: run
spec:
 	alias:
 	- cluster-1
 	- cluster-2
 	commands:
 	- kubectl exec mypod -- somecommand
 	- kubectl logs mypod

**Reply by richardcase** [reply]
I like the idea of using a declarative config for this. Thats a lot better.

**Reply by ramnar** [reply]
Both are really nice approaches to solve the problem.First one is easy to use for single command and second one for multiple commands. I will be happy to contribute to this feature, whichever approach we finalize on.

**Reply by ramnar** [reply]
I was thinking about it. As yaml is a one time configuration, second approach looks great

**Reply by richardcase** [reply]
Lets go with the yaml approach then. That would be great if you wanted to contribute this @ramnar . Let me know if you need any help getting started.

**Comment by richardcase** [comment]
@ramnar - i have created an issue for this: #228 . If you want to pick this up could you comment on the issue to say you want to do it and i'll assign it to you.
